fenix/app/src/test/java/org/mozilla/fenix/browser/TranslationsIconBindingTest.kt

```diff
@@ -0,0 +1,168 @@
+import mozilla.components.support.test.rule.runTestOnMain
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.mockito.Mockito.spy
+
+@RunWith(AndroidJUnit4::class)
+class TranslationsIconBindingTest {
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
+
+    lateinit var browserStore: BrowserStore
+
+    private val tabId = "1"
+    private val tab = createTab(url = tabId, id = tabId)
+    private val onIconChanged: (
+        isVisible: Boolean,
+        isTranslated: Boolean,
+        fromSelectedLanguage: Language?,
+        toSelectedLanguage: Language?,
+    ) -> Unit = spy()
+
+    @Test
+    fun `GIVEN translationState WHEN translation status isTranslated THEN invoke onIconChanged callback`() =
+        runTestOnMain {
+            val englishLanguage = Language("en", "English")
+            val spanishLanguage = Language("es", "Spanish")
+
+            browserStore = BrowserStore(
+                BrowserState(
+                    tabs = listOf(tab),
+                    selectedTabId = tabId,
+                ),
+            )
+
+            val binding = TranslationsIconBinding(
+                browserStore = browserStore,
+                sessionId = tabId,
+                onIconChanged = onIconChanged,
+            )
+            binding.start()
+
+            val detectedLanguages = DetectedLanguages(
+                documentLangTag = englishLanguage.code,
+                supportedDocumentLang = true,
+                userPreferredLangTag = spanishLanguage.code,
+            )
+
+            val translationEngineState = TranslationEngineState(
+                detectedLanguages = detectedLanguages,
+                error = null,
+                isEngineReady = true,
+                requestedTranslationPair = TranslationPair(
+                    fromLanguage = englishLanguage.code,
+                    toLanguage = spanishLanguage.code,
+                ),
+            )
+
+            val supportLanguages = TranslationSupport(
+                fromLanguages = listOf(englishLanguage),
+                toLanguages = listOf(spanishLanguage),
+            )
+
+            browserStore.dispatch(
+                TranslationsAction.SetSupportedLanguagesAction(
+                    tabId = tab.id,
+                    supportedLanguages = supportLanguages,
+                ),
+            ).joinBlocking()
+
+            browserStore.dispatch(
+                TranslationsAction.TranslateStateChangeAction(
+                    tabId = tabId,
+                    translationEngineState = translationEngineState,
+                ),
+            ).joinBlocking()
+
+            browserStore.dispatch(
+                TranslationsAction.TranslateSuccessAction(
+                    tabId = tab.id,
+                    operation = TranslationOperation.TRANSLATE,
+                ),
+            ).joinBlocking()
+
+            Mockito.verify(onIconChanged).invoke(
```

Comment:
Thanks for the tests! Could you also add the parameter names here and the tests below to improve readability?

Reply:
I can't add names to the callback invoke function.